src/Course/List.hs

```diff
@@ -261,12 +239,11 @@ seqOptional =
 --
 -- >>> find (const True) infinity
 -- Full 0
-find ::
-  (a -> Bool)
-  -> List a
-  -> Optional a
-find =
-  error "todo: Course.List#find"
+find :: (a -> Bool) -> List a -> Optional a
+find p = maybeFirst . filter p
```

Comment:
very nice :)

I think there's already `headOr`  above, so can:

`find p = headOr Nil . filter p`

... if you like.

Reply:
This utilises laziness to make it efficient 👍